package-res/resources/web/pentaho/type/changes/ComplexChangeset.js

```diff
@@ -0,0 +1,247 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+define([
+  "./Changeset",
+  "./ListChangeset",
+  "./Replace",
+  "../../util/object",
+  "../../util/error"
+], function(Changeset, ListChangeset, Replace, O, error) {
+
+  "use strict";
+
+  /**
+   * @name ComplexChangeset
+   * @memberOf pentaho.type.changes
+   * @class
+   * @extends pentaho.type.changes.Changeset
+   * @amd pentaho/type/changes/ComplexChangeset
+   *
+   * @classDesc The class `ComplexChangeset` describes a set of changes to the values of properties
+   * in a [complex]{@linkplain pentaho.type.Complex} value.
+   *
+   * @constructor
+   * @description Creates a new instance.
+   *
+   * @param {!pentaho.type.Complex} owner - The complex value where the changes take place.
+   */
+  return Changeset.extend("pentaho.type.changes.ComplexChangeset", /** @lends pentaho.type.changes.ComplexChangeset#*/{
+
+    constructor: function(owner) {
+      this.base(owner);
+
+      this._changes = {};
+    },
+
+    /**
+     * Gets the complex value where the changes take place.
+     *
+     * @name pentaho.type.changes.ComplexChangeset#owner
+     * @type {!pentaho.type.Complex}
+     * @readonly
+     */
+
+    //region public interface
+    /**
+     * Gets the type of change.
+     *
+     * @type {string}
+     * @readonly
+     * @default "complex"
+     */
+    get type() {
+      return "complex";
+    },
+
+    /**
+     * Gets a value that indicates if this changeset contains any changes.
+     *
+     * @type {boolean}
+     */
+    get hasChanges() {
+      var hasChanges = false;
+
+      O.eachOwn(this._changes, function(change) {
+        if(!(change instanceof Changeset) || change.hasChanges) {
+          hasChanges = true;
+          return false;
+        }
+      });
+
+      return hasChanges;
+    },
+
+    _clearChanges: function() {
+
+      O.eachOwn(this._changes, function(change) {
+        if(change instanceof Changeset) change.cancel();
+      });
+
+      this._changes = {};
+    },
+
+    /**
+     * Gets the [change]{@link pentaho.type.changes.ValueChange} object associated with the specified property.
+     *
+     * @param {nonEmptyString|!pentaho.type.Property.Type} name - The property name or type object.
```

Comment:
I can't agree, but I do see why you suggested this.

"name" is either a string, or an object that describes that property. 
We are abusing the language here and calling it a "type object". The expression "type of object" is most likely to be understood as the "type constructor", i.e. the function that returns a new object upon invocation.